nats/bench_command.go

```diff
@@ -64,37 +113,123 @@ func (c *benchCmd) bench(_ *kingpin.ParseContext) error {
 		return fmt.Errorf("number of messages should be greater than 0")
 	}
 
-	log.Printf("Starting benchmark [msgs=%s, msgsize=%s, pubs=%d, subs=%d]", humanize.Comma(int64(c.numMsg)), humanize.IBytes(uint64(c.msgSize)), c.numPubs, c.numSubs)
+	log.Printf("Starting benchmark [msgs=%s, msgsize=%s, pubs=%d, subs=%d, js=%v, storage=%s, syncpub=%v, pubbatch=%s, pull=%v, pullbatch=%s, request=%v, reply=%v, noqueue=%v, maxackpending=%s, replicas=%d, nopurge=%v, nodelete=%v]", humanize.Comma(int64(c.numMsg)), humanize.IBytes(uint64(c.msgSize)), c.numPubs, c.numSubs, c.js, c.storage, c.syncPub, humanize.Comma(int64(c.pubBatch)), c.pull, humanize.Comma(int64(c.pullBatch)), c.request, c.reply, c.noQueueGroup, humanize.Comma(int64(c.maxAckPending)), c.replicas, c.noPurge, c.noDeleteStream)
 
-	if c.ack && c.progress {
-		log.Printf("Disabling progress bars in request mode")
-		c.progress = false
+	if c.numPubs == 0 && c.numSubs == 0 {
+		log.Fatalf("You must have at least one publisher or at least one subscriber... try adding --pub 1 and/or --sub 1 to the arguments")
```

Comment:
so wouldnt the previous default of 1 on pub make more sense? That way the default happy path is to public, if you really want to set sub only you can disable that. The tool should function with as few as possible flags

Reply:
I took that default out because I really didn't like when you would use

`nats bench testsubject --sub 1 ...` 

It would then end up creating a subscriber _and_ a publisher (which can definitely surprise you)

So you end up always having to specify `--pub=0` if you do not want to run a publisher. And in this new version of nats bench most of the time you will run it with either no subscriber(s) or no publisher(s) (while before since no JS mode of operation you would maybe use both a publisher and a subscriber more often - which BTW I don't consider a very realistic scenario: in 'real life' your application usually don't just subscribe to their own publications)

I agree that you want to keep the number of required flags as few as you can, but in this case having that pub default to 1 and sub default to 1, besides sometimes surprising you, means there are many times you need to add a `--pub 0` argument (so actually it means the old way would cause you to have to add more flags than now, most of the time since now you most of the time would not run both a publisher and a subscriber flags at the same time).

I really don't want to change what I did, once you get used to it a bit it (to me at least) makes more sense to have both --pub and --sub default to 0 (than having just one of them default to 0 and the other to 1 like it was)

Reply:
OK